lib/ui/screens/thread.dart

```diff
@@ -1,57 +1,69 @@
 import 'package:flutter/material.dart';
+import 'package:reddit_app/persistence/DTO/comment_dto.dart';
+import 'package:reddit_app/persistence/DTO/post_dto.dart';
+import 'package:reddit_app/persistence/DTO/user_dto.dart';
+import 'package:reddit_app/persistence/DTO/subreddit_dto.dart';
+import 'package:reddit_app/persistence/persistence_adapter.dart';
+import 'package:reddit_app/persistence/persistence_port.dart';
 import 'package:reddit_app/ui/widgets/comments/comment.dart';
 import 'package:reddit_app/ui/widgets/posts/post_extended.dart';
 import 'package:reddit_app/ui/widgets/scaffold/footer.dart';
 
 class Thread extends StatefulWidget {
-  const Thread({super.key});
+  final int post;
+
+  const Thread({super.key, required this.post});
 
   @override
-  State<Thread> createState() => _ThreadState();
+  // ignore: no_logic_in_create_state
+  State<Thread> createState() => _ThreadState(post);
```

Comment:
Don't do this.
You can get the post through widget.post instead of passing it in the createState()